traverse_test.go

```diff
@@ -46,3 +47,262 @@ func TestNetworksWithInvalidSearchTree(t *testing.T) {
 	assert.NotNil(t, n.Err(), "no error received when traversing an broken search tree")
 	assert.Equal(t, n.Err().Error(), "invalid search tree at 128.128.128.128/32")
 }
+
+func TestNetworksWithinV4SearchInV4Db(t *testing.T) {
+	var network = &net.IPNet{IP: make(net.IP, 4), Mask: net.CIDRMask(0, 32)}
+
+	for _, recordSize := range []uint{24, 28, 32} {
+		fileName := testFile(fmt.Sprintf("MaxMind-DB-test-ipv4-%d.mmdb", recordSize))
+		reader, err := Open(fileName)
+		require.Nil(t, err, "unexpected error while opening database: %v", err)
+		defer reader.Close()
+
+		n := reader.NetworksWithin(network)
+		var innerIPs []string
+
+		for n.Next() {
+			record := struct {
+				IP string `maxminddb:"ip"`
+			}{}
+			network, err := n.Network(&record)
+			assert.Nil(t, err)
+			assert.Equal(t, record.IP, network.IP.String(),
+				"expected %s got %s", record.IP, network.IP.String(),
+			)
+			innerIPs = append(innerIPs, network.String())
+		}
+
+		expectedIPs := []string{
+			"1.1.1.1/32",
+			"1.1.1.2/31",
+			"1.1.1.4/30",
+			"1.1.1.8/29",
```

Comment:
I think it would be good to have a test that selects multiple records, but only a subset of the ones within an address space. For instance, `81.2.69.128/26` in the GeoIP2 Country test database.